azkaban-common/src/main/java/azkaban/executor/container/watch/AzPodStatusDrivingListener.java

```diff
@@ -44,21 +46,26 @@
  * database status updates.
  */
 @Singleton
-public class AzPodStatusDriver implements RawPodWatchEventListener {
-  private static final Logger logger = LoggerFactory.getLogger(AzPodStatusDriver.class);
+public class AzPodStatusDrivingListener implements RawPodWatchEventListener {
+  private static final Logger logger = LoggerFactory.getLogger(AzPodStatusDrivingListener.class);
 
-  private static final int THREAD_POOL_SIZE = 4;
-  public static final int SHUTDOW_TERMINATION_TIMEOUT_SECONDS = 5;
+  private static final int DEFAULT_THREAD_POOL_SIZE = 4;
+  public static final int SHUTDOWN_TERMINATION_TIMEOUT_SECONDS = 5;
+  private final int threadPoolSize;
   private final ExecutorService executor;
   private final ImmutableMap<AzPodStatus, List<Consumer<AzPodStatusMetadata>>> listenerMap;
 
   /**
-   * Create a new instance of {@link AzPodStatusDriver}.
+   * Create a new instance of {@link AzPodStatusDrivingListener}.
    */
   @Inject
-  public AzPodStatusDriver() {
-    // Worth considering if we should get the thread-pool size from a config property instead.
-    executor = Executors.newFixedThreadPool(THREAD_POOL_SIZE,
+  public AzPodStatusDrivingListener(Props azkProps) {
```

Comment:
Using annotation @NonNull would be better.

Reply:
Discussed in a previous comment.